Tests/Unit/MSFT_xDefaultGatewayAddress.Tests.ps1

```diff
@@ -50,193 +80,259 @@ InModuleScope MSFT_xDefaultGatewayAddress {
                 $Result.Address | Should BeNullOrEmpty
             }
         }
-
     }
 
-    Describe 'Test-Properties' {
+    #######################################################################################
 
-        Context 'invoking without -Apply switch and default gateway is exists' {
+    Describe 'Set-TargetResource' {
 
-            #region Mocks
-            Mock Get-NetRoute -MockWith {
-                [PSCustomObject]@{
-                    NextHop = '192.168.0.1'
-                    DestinationPrefix = '0.0.0.0/0'
-                    InterfaceAlias = 'Ethernet'
-                    InterfaceIndex = 1
-                    AddressFamily = 'IPv4'
-                }
+        #region Mocks
+        Mock Get-NetAdapter -MockWith { [PSObject]@{ Name = 'Ethernet' } }
+
+        Mock Get-NetRoute -MockWith {
+            [PSCustomObject]@{
+                NextHop = '192.168.0.1'
+                DestinationPrefix = '0.0.0.0/0'
+                InterfaceAlias = 'Ethernet'
+                InterfaceIndex = 1
+                AddressFamily = 'IPv4'
             }
-            #endregion
+        }
+
+        Mock Remove-NetRoute
 
-            It 'when default gateway does not match should be $false' {
+        Mock New-NetRoute
+
+        #endregion
+
+        Context 'invoking with no Default Gateway Address' {
+            It 'should rerturn $null' {
```

Comment:
Thanks, helps that we've scopped this one down a bit more so its easier to comb over. :laughing: